modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -627,13 +629,43 @@ private void parseNullSafe(ExtensionParameter extensionParameter, ParameterDecla
         @Override
         protected void visitBasicType(MetadataType metadataType) {
           throw new IllegalParameterModelDefinitionException(format("Parameter '%s' is annotated with '@%s' but is of type '%s'. That annotation can only be "
-              + "used with complex types (Pojos, Lists, Maps)",
+              + "used with complex instantiable types (Pojos, Lists, Maps)",
                                                                     extensionParameter.getName(), NullSafe.class.getSimpleName(),
                                                                     extensionParameter.getType().getName()));
         }
       });
 
-      parameter.withModelProperty(new NullSafeModelProperty());
+      Class<?> defaultType = extensionParameter.getAnnotation(NullSafe.class).get().defaultImplementingType();
+
+
+      if (isInstantiable(parameter.getDeclaration().getType()) && !defaultType.isAssignableFrom(Object.class)) {
```

Comment:
it's not clear in this code where and how are you accounting for the cases in which:

* The default type is not needed at all
* the default type is needed but not specified